### DISCUSSION TITLE
Do I need an Output

### DISCUSSION THREAD

**Discussion by dlionz** [dbosoft/YaNco#193] [Q&A] [answered]
Hello all, so RFC is something very new to me. I want to post value to my clients SAP table. Do I need to have an Output property as part of my post? I think the error catch will be sufficient enough but I dont need my post action to return anything. 

 ```
using (var con = new Dbosoft.YaNco.RfcContext(connFunc))
            {
                    // fake
                    await con.CallFunction("FUNC_NAME",
                        Input: f => f.SetField("PROJECT", 12345)
                    l => Console.WriteLine($"Error: {l.Message}"));
                    con.Dispose();
            };
```

**Comment by fw2568** [comment, marked as answer]
Hi,

if you need no output you can use CallFunctionOneWay. This method expects
only input but still returns in Error state if failed.

Best Regards,
Frank

**Reply by dlionz** [reply]
oh that sounds perfect, Ill look at that thanks!

**Reply by dlionz** [reply]
Do you have an exmaple that I can look at?

**Reply by fw2568** [reply]
mmhm, not really, but taking your code it would like like this:

``` csharp
using (var con = new Dbosoft.YaNco.RfcContext(connFunc))
           {
                   // fake
                   await con
                    .CallFunctionOneWay("FUNC_NAME", f => f.SetField("PROJECT", 12345))
                    .IfLeft(l => Console.WriteLine($"Error: {l.Message}"));
                   
                     // no need to dispose, its in using section
                     // con.Dispose();  
           };
```

IfLeft is like Match, but only handles left (=Error) results.

**Reply by dlionz** [reply]
Hey Frank, so I was able to write it like this and the project will build but I'm not sure this will work.
```
using (var con = new Dbosoft.YaNco.RfcContext(connFunc))
{
     con.CallFunctionOneWay("FUNC_NAME",
                        Input: f => f.SetField("PROJECT", 12345));
}
```

I tried yours above and I was receiving this error. "cannot convert lambda expression to type cancelationToken because it is not a delegate type."

**Reply by fw2568** [reply]
sorry, I overlooked a ). See updated sample. 
Your version will compile, but contains no error handling and no await.

**Reply by dlionz** [reply]
awesome! thank you so much for your help
